compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -649,29 +649,334 @@ abstract class GenJSCode extends plugins.PluginComponent
+            case js.Block(stats) =>
+              stats.dropWhile {
+                case js.ApplyStatic(_, mtd, _) if isConstructor(mtd.name) => false
+                case _                                                    => true
+              }.tail
+
+            case _ => Nil
+          }
+        }
+        js.Block(postSuperCall :+ mkSubPostCalls(overrideNumRef, isConstructor))
+      }
+
+      private def mkOverrideNumsCond(numRef: js.VarRef,
+          numBounds: (Int, Int))(implicit pos: Position) = numBounds match {
+        case (lo, hi) if lo == hi =>
+          js.BinaryOp(js.BinaryOp.===, js.IntLiteral(lo), numRef)
+
+        case (lo, hi) if lo == hi - 1 =>
+          val lhs = js.BinaryOp(js.BinaryOp.===, numRef, js.IntLiteral(lo))
+          val rhs = js.BinaryOp(js.BinaryOp.===, numRef, js.IntLiteral(hi))
+          js.If(lhs, js.BooleanLiteral(true), rhs)(jstpe.BooleanType)
+
+        case (lo, hi) =>
+          val lhs = js.BinaryOp(js.BinaryOp.Num_<=, js.IntLiteral(lo), numRef)
+          val rhs = js.BinaryOp(js.BinaryOp.Num_<=, numRef, js.IntLiteral(hi))
+          js.BinaryOp(js.BinaryOp.Boolean_&, lhs, rhs)
+          js.If(lhs, rhs, js.BooleanLiteral(false))(jstpe.BooleanType)
+      }
+    }
+
+    private def isCallToContructor(cls: jstpe.ClassType,
+        methodName: String): Boolean = {
+      ir.Definitions.isConstructorName(methodName) && cls == currentClassType
+    }
+
+    private def zipMap[T, U, V](xs: List[T], ys: List[U])(
+        f: (T, U) => V): List[V] = {
+      for ((x, y) <- xs zip ys) yield f(x, y)
+    }
+
+    /** mkOverloadSelection return a list of `stats` with that starts with:
+     *  1) The definition for the local variable that will hold the overload
+     *     resolution number.
+     *  2) The definitions of all local variables that are used as parameters
+     *     in all the constructors.
+     *  3) The overload resolution match/if statements. For each overload the
+     *     overload number is assigned and the parameters are cast and assigned
+     *     to their corresponding variables.
+     */
+    private def mkOverloadSelection(constructorTree: ConstructorTree,
+        overloadIdent: js.Ident, dispatchResolution: js.Tree,
+        isConstructor: Set[String])(implicit pos: Position): List[js.Tree]= {
+      if (!constructorTree.hasSubConstructors) {
+        dispatchResolution match {
+          case js.Block(stats) =>
+            val js.ApplyStatic(_, method, _) = stats.last
+            val refs = constructorTree.getParamRefs(method.name)
+            val rhss = stats.collect {
+              case js.VarDef(_, _, _, rhs) => rhs
+            }
+            zipMap(refs, rhss) { (ref, rhs) =>
+              js.VarDef(ref.ident, ref.tpe, mutable = false, rhs)
+            }
+
+          case js.ApplyStatic(_, mtd, js.This() :: Nil)
+              if ir.Definitions.isConstructorName(mtd.name) =>
+            Nil
+
+          case tree =>
+            List(tree)
```

Comment:
Can this happen? It seems this is asking for silently ignoring cases where we don't have the shape we expected.